docs/src/userguide/language_basics.rst

```diff
@@ -404,20 +637,56 @@ returns small results.
 
 If all possible return values are legal and you
 can't reserve one entirely for signalling errors, you can use an alternative
-form of exception value declaration::
+form of exception value declaration
 
-    cdef int spam() except? -1:
-        ...
+.. tabs::
 
-The "?" indicates that the value ``-1`` only signals a possible error. In this
-case, Cython generates a call to :c:func:`PyErr_Occurred` if the exception value
-is returned, to make sure it really received an exception and not just a normal
-result.
+    .. group-tab:: Pure Python
 
-There is also a third form of exception value declaration::
+        .. code-block:: python
 
-    cdef int spam() except *:
-        ...
+            @exceptval(-1, check=True)
+            def spam() -> cython.int:
+                ...
+
+        The ``check=True`` indicates that the value ``-1`` only signals a possible error.
+        In this
+        case, Cython generates a call to :c:func:`PyErr_Occurred` if the exception value
+        is returned, to make sure it really received an exception and not just a normal
+        result.
+
+    .. group-tab:: Cython
+
+        .. code-block:: cython
+
+            cdef int spam() except? -1:
+                ...
+
+        The ``?`` indicates that the value ``-1`` only signals a possible error.
+        In this
+        case, Cython generates a call to :c:func:`PyErr_Occurred` if the exception value
+        is returned, to make sure it really received an exception and not just a normal
+        result.
+
+
+There is also a third form of exception value declaration
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            @exceptval(check=True)
```

Comment:
You forgot `cython.cfunc` and same in other places

Reply:
But I doubt it a little, so you better double check

Reply:
Done thank you!